### DISCUSSION TITLE
Flutter iOS .ipa file generation failing

### DISCUSSION THREAD

**Discussion by akmalahmed525** [codemagic-ci-cd/codemagic-docs#2113] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. I have configured 3 different flavours (Dev, Prod and QA) with Firebase
2. Xcode version `15.0`, Cocoapods version `1.13.0`
3. Start a new iOS build

### Expected results

The commands in the below `yaml` snippet should run successfully and it should create an `.ipa` file

```yaml

scripts:
 ...
 ...
 ...
 - name: Install pods
         script: |
           find . -name "Podfile" -execdir pod install \;
 - name: Increment build number & build ipa file
         script: |
           CURRENT_BUILD_NUMBER=$(app-store-connect get-latest-testflight-build-number $APPLE_ID)
           NEW_BUILD_NUMBER=$(($CURRENT_BUILD_NUMBER + 1))
           flutter build ipa --release --build-number=$NEW_BUILD_NUMBER --export-options-plist ~/export_options.plist --flavor $XCODE_SCHEME -t $ENTRY_POINT
 ...
 ...
 ...
```

### Actual results

Throws the following error for the `find . -name "Podfile" -execdir pod install \;` command. It was working perfectly for the previous builds for other flavours.

```
[!] Invalid `Podfile` file: /Users/builder/clone/ios/Flutter/Generated.xcconfig must exist. If you're running pod install manually, make sure flutter pub get is executed first.

 #  from /Users/builder/clone/ios/Podfile:16
 #  -------------------------------------------
 #    unless File.exist?(generated_xcode_build_settings_path)
 >      raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
 #    end
 #  -------------------------------------------
```

Which fails the following script after that ` Increment build number & build ipa file`

```

Running pod install...                                             40.1s
Running Xcode build...                                          
Xcode archive done.                                         124.2s
Failed to build iOS app
Error (Xcode): /Users/Desktop/config: No such file or directory

Encountered error while archiving for device.


Build failed :|
Step 8 script `Increment build number & build ipa file` exited with status code 1
```

### Build id (optional)

6527d3426e1a23c6c3b374be

**Comment by himesh-cm** [comment]
Hi @akmalahmed525, Can you run a build with `Xcode 14.3.1` for debugging purposes and share the build id?

Also, Please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

**Comment by dtrdic** [comment, marked as answer]
We resolved this issue on intercom but i'll post here as well.. the root cause of the issue was identified in this error message:

```
[!] Invalid `Podfile` file: /Users/builder/clone/ios/Flutter/Generated.xcconfig must exist. If you're running pod install manually, make sure flutter pub get is executed first.

 #  from /Users/builder/clone/ios/Podfile:16
 #  -------------------------------------------
 #    unless File.exist?(generated_xcode_build_settings_path)
 >      raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
 #    end
 #  -------------------------------------------
```
The issue stemmed from the user having different flavors, and the ios/Runner.xcodeproj/project.pbxproj file was referencing the incorrect configuration path location, leading to the `Generated.xcconfig must exist` error message.

**Reply by francescorinaldi** [reply]
Hi @dtrdic, I got the same issue. How can I solve it? Thanks!

**Reply by isacmoura** [reply]
So, what was the solution for this?

**Comment by vinisxs** [comment]
This problem appeared to me after I upgraded my project to Flutter 3.27. I managed to resolve it by running `flutter build apk --config-only` **before** the `pod install`. As strange as it seems, the build apk command generates the missing file.
